src/main/java/com/icthh/xm/tmf/ms/customer/lep/keyresolver/ProfileKeyResolver.java

```diff
@@ -17,4 +24,27 @@
         String translated = translateToLepConvention(profile);
         return new String[]{translated};
     }
+
+    @Override
+    protected <T> T getRequiredParam(LepMethod method, String paramName, Class<T> valueType) {
+        return ofNullable(getParamValue(method, paramName, valueType))
+            .or(() -> getHeaderValue(paramName, valueType))
+            .orElseThrow(() -> new IllegalArgumentException(
+                String.format("Neither LEP method %s nor request header contain parameter %s",
+                    getMethodDescription(method), paramName)));
+    }
+
+    private <T> Optional<T> getHeaderValue(final String headerName, final Class<T> valueType) {
+        return ofNullable((ServletRequestAttributes) RequestContextHolder.getRequestAttributes())
+            .map(attributes -> attributes.getRequest().getHeader(headerName))
+            .map(valueType::cast);
+    }
+
+    private String getMethodDescription(LepMethod method) {
+        MethodSignature methodSignature = method.getMethodSignature();
```

Comment:
please consider rewriting method body in a functional way: 
```
ofNullable(method.getMethodSignature())
            .map(MethodSignature::getMethod)
            .map(Method::toString)
            .orElseGet(method::toString);
```

Reply:
Done